src/event/metric.rs

```diff
@@ -360,6 +379,112 @@ impl Display for Metric {
     }
 }
 
+impl Object for Metric {
+    fn insert(&mut self, path: &remap::Path, value: remap::Value) -> Result<(), String> {
+        if path.is_root() {
+            return Err("cannot set root path for a metric event".into());
+        }
+
+        match path.segments() {
+            [Segment::Field(tags), Segment::Field(field)] if tags.as_str() == "tags" => {
+                let value = value.as_bytes().ok_or("value must be a string")?;
```

Comment:
We have `try_bytes()` for this, which gives a more consistent error (e.g. "value must be string, but got ...")

(similar for all the other `ok_or`'s below)